runtime/gc_vlhgc/IncrementalGenerationalGC.cpp

```diff
@@ -1941,11 +1945,9 @@ MM_IncrementalGenerationalGC::mainThreadConcurrentCollect(MM_EnvironmentBase *en
 	/* note that we can't check isConcurrentWorkAvailable at this point since another thread could have set _forceConcurrentTermination since the
 	 * main thread calls this outside of the control monitor
 	 */
-	Assert_MM_true(NULL == env->_cycleState);
```

Comment:
Instead of removing it completely, we change this to assert `env->_cycleState == _persistentGlobalMarkPhaseState;`